syntax = "proto3";

// Request to fetch historical trade data.
message SecurityHistoricalRequest {
    string file_path = 1;
    uint32 security_id = 2;
}

message CurrencyHistoricalRequest {
    string file_path = 1;
    string exchange = 2;
    string bank = 3;
}

// Response to the request of an instrument's history
message InstrumentHistoricalResponse {
    double price = 1;
    string time = 2; // Send "DNE" if null
}

// Historical service definition
service HistoricalSimulation {
     // Streams historical data for a particular security
     rpc StreamSecurityHistory (SecurityHistoricalRequest) returns (stream InstrumentHistoricalResponse) {}

     // Streams historical currency data back
     rpc StreamCurrencyHistory (CurrencyHistoricalRequest) returns (stream InstrumentHistoricalResponse) {}
}